### PR TITLE
Fix userdata derivatives for interpolated params on GPU

### PR DESCRIPTION
**Pull request by lecocqp**
## Description

Following #1657, it turns out we forgot to zero initialize the userdata derivatives for the optix path. 
This leads to some rendering glitches caused by the memory garbage left in those derivatives.

## Checklist:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [ ] I have previously submitted a [Contributor License Agreement](http://opensource.imageworks.com/cla/).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite (adding new test cases if necessary).
- [x] My code follows the prevailing code style of this project.

### CODE REVIEW

**Comment by lgritz** [comment]
Pascal, you can fix the DCO by adding a sign-off:

    git commit --amend -s
    git push origin <branchname> --force

For the EasyCLA, it seems you weren't on the SPI approved list. (How did that happen? Is this really your first OSL PR?) Anyway, I added you, so you should see something to click that will let you accept it and they it will be cleared for this and any future PRs.

**Comment by lgritz** [comment]
Don't worry about the two test failures for icc and icx -- that's just Intel's servers being flaky and the CI runs unable to download those compilers. It comes and goes.

**Review by lgritz — CHANGES_REQUESTED** [review]
The code change LGTM, thanks for the fix.
I need the CLA and DCO fixed before I can merge.

**Comment by lecocqp** [comment]
Yes, very first OSL PR, I only contributed to OIIO so far. For the EasyCLA should I proceed as a Corporate or Individual contributor?

**Comment by lgritz** [comment]
Corporate, tell it you're with Sony Imageworks, I've already put you on our list.

**Comment by lecocqp** [comment]
I completed the CLA authorization request. Is there something else I need to do for getting the CLA failure resolved? Another push force commit?

**Comment by lgritz** [comment]
I'm not sure, I'll try to figure it out.

**Comment by lgritz** [comment]
Hmmm, can you just diddle it and re-push:

    git commit --amend       # force a sha change, don't edit anything
    git push origin blah --force

maybe that will get it unstuck.

**Comment by lecocqp** [comment]
I re-pushed the branch but same thing. The EasyCLA process says this: "The Corporate CLA process requires you to be approved by your organization's CLA Manager. A CLA Manager at your organization will receive your request via email immediately after you submit it. To expedite the process, you can follow up with them directly.".

Maybe you (or someone else?) have receive an email for approving the authorization?

**Comment by lgritz** [comment]
I am the CLA manager for the company. I see you already on the approved list, and there are no additional requests waiting for my approval. If the rest of the CI passes, I may be able to just force a merge and override the check. I just don't know why EasyCLA is still giving me trouble. @jmertic do you have any insights?

**Comment by lgritz** [comment]
Pascal, is it possible that the email associated with the commit doesn't match the email associated with your github account?

**Comment by lecocqp** [comment]
Oh you're right! I didn't realize that the commit was signed off with my Imageworks email instead of my gmail address.  I'll look into it tomorrow.

**Comment by lgritz** [comment]
I think you should be able to fix with

    git commit --amend --author "Pascal Lecocq <your@email>" -s

and then re-push --force

But also, I just added your work email to the permissions list, so it may just work with another force push? (Though changing the email to be consistent will make it so that future PRs all look like the same person instead of you appearing under multiple aliases.)

**Comment by lecocqp** [comment]
I setup my personal email for that specific repo in the .git/config using a [user] tag. I just pushed another commit, let's see how it goes.

**Review by lgritz — APPROVED** [review]
Code LGTM. I know you're authorized under our CLA, so I'm going to force the merge even if it doesn't work this time. We'll figure it out next time if it's still not working (it may be fine if a PR is first started after you have been added and with the emails consistent).

**Pull request merged**
